### PR TITLE
[s] Makes used grenades slightly more obvious. Also prevents DOSing admins.

### LINKED ISSUE #64640: Spamming grenades without a payload floods admin chat with messages and can crash them.
## Reproduction:
Step one make a non explosive grenade.
Step two spam activate it. (Although thats not how this incident happened its roughly the same steps)

Step three....

sentientgrenadesp2
sentientgrenadesp1

I`m not sure if i should include this in the same issue report but I wasnt actually the one triggering the circuit grenades, I was turning posibrains into sentient plushies by putting them into an assembly circuit and then attaching them to a grenade and giving those grenades a nonlethal payload that they could activate, what ended up happening was the posibrain players used their payload on the evac shuttle then got bored and started spamming the detonate key and then I got bwoinked.

Considering that just a player spamming a keypress can be enough to crash the admins, you could very easily outdo that 100x and make an admin ddos machine by binding 2 circuit radio modules to eachother in a loop and having their outputs activate a grenade which I would assume would be pretty bad so I would say that a fix for this should be high priority.

### PR DESCRIPTION
**Pull request by TemporalOroboros**
Adds a variable to tell whether a grenade can't detonate properly.
Examining can be used to tell if the grenade has already been used.
A flag on it is set when the grenade detonates so that used grenades can't be reused until it's reset.
It also prevents spamming admins with grenade primed messages since there's no point if the grenade won't work.

## About The Pull Request
Adds a variable to track whether a grenade has already been used.
If it has it prevents the grenade from detonating again until the grenade has been reset. (Resetting only implemented for chemical grenades.)
It also prevents priming the grenade/pulsing the grenades wires from messaging admins (but still lets them log) because there's no point in telling the admins if the grenade isn't going to do anything.
The status of the grenade shows up in the examine text.

## Why It's Good For The Game
1. Makes it slightly more obvious that the cleaner grenades in the halls have already been used.
2. Prevents DOSing admins with grenade logging spam. (Fixes #64640)

## Changelog

:cl:
qol: Used grenades now look used when examined.
fix: It should no longer be possible to DC admins by spamming grenades.
/:cl:

### CODE REVIEW

**Comment by TheVekter** [comment]
Based, this should prevent this from happening.

image

**Review by tralezab — CHANGES_REQUESTED** [review]
the rest looks good

File: code/game/objects/items/grenades/_grenade.dm [hunk 1]
```diff
@@ -17,6 +23,8 @@
 	slot_flags = ITEM_SLOT_BELT
 	resistance_flags = FLAMMABLE
 	max_integrity = 40
+	/// Whether this grenade can detonate.
+	var/dud = FALSE
```
[the remark this answers is not in the document]
Comment (TemporalOroboros) [reply code/game/objects/items/grenades/_grenade.dm]: Done.

**Comment by Elli-Skala** [comment]
This takes into account the advanced release grenade right?

**Pull request merged**